Metadata API: linting _type is an issue for API users

Any code that uses the Metadata API is going to get linting results like this: 
```
file.py:299:11: W0212: Access to a protected member _type of a client class (protected-access)
file.py:301:41: W0212: Access to a protected member _type of a client class (protected-access)
file.py:336:11: W0212: Access to a protected member _type of a client class (protected-access)
file.py:338:46: W0212: Access to a protected member _type of a client class (protected-access)
file.py:396:11: W0212: Access to a protected member _type of a client class (protected-access)
file.py:398:45: W0212: Access to a protected member _type of a client class (protected-access)
```

`Signed._type` is named like it is to copy the file format 1-to-1 which makes some sense...  but this is also pretty unacceptable. Maybe we should make another property `type` that works exactly like `_type` ?

--- a/tuf/api/metadata.py
+++ b/tuf/api/metadata.py
@@ -315,20 +315,24 @@ class Signed:
     are common for all TUF metadata types (roles).
 
     Attributes:
-        _type: The metadata type string.
+        _type: The metadata type string. Also available without underscore.
         version: The metadata version number.
         spec_version: The TUF specification version number (semver) the
             metadata format adheres to.
         expires: The metadata expiration datetime object.
         unrecognized_fields: Dictionary of all unrecognized fields.
-
     """
 
     # Signed implementations are expected to override this
     _signed_type = None
 
+    # _type and type are identical: 1st replicates file format, 2nd passes lint
     @property
     def _type(self):
+        return self._signed_type
+
+    @property
+    def type(self):
         return self._signed_type
 
     # NOTE: Signed is a stupid name, because this might not be signed yet, but

